### PR TITLE
[GraphBolt] Datapipe for copying to given device

### PR DESCRIPTION
**Pull request by BarclayII**
## Description
This is a sub-PR of #5901 .

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR

### CODE REVIEW

File: python/dgl/graphbolt/copy_to.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+from torchdata.datapipes.iter import IterDataPipe
+
+from ..utils import recursive_apply
+
+
+def _to(x, device):
+    return x.to(device) if hasattr(x, "to") else x
+
+
+class CopyTo(IterDataPipe):
```
Comment (frozenbugs) [inline python/dgl/graphbolt/copy_to.py:13]: Why not also use Mapper?

Comment (BarclayII) [reply python/dgl/graphbolt/copy_to.py:13]: I think the current implementation is more straightforward.  If I use a `Mapper` then I'll have to write it like:
```python
def __init__(self, dp, device):
    self.device = device
    super().__init__(dp, self.map_func)
def map_func(self, data):
    return recursive_apply(data, _to, self.device)
```

Comment (frozenbugs) [reply python/dgl/graphbolt/copy_to.py:13]: This is 5 lines compare to existing 8 lines of code, it is not bad.

Comment (BarclayII) [reply python/dgl/graphbolt/copy_to.py:13]: But I don't think it's as readable as 8 lines of code.  A for loop is cleaner than passing in a function IMO.

**Review by frozenbugs — APPROVED** [review]

**Pull request merged**